Tests/ApodiniTests/SharedSemanticModelBuilderTests.swift

```diff
@@ -149,4 +166,18 @@ final class SharedSemanticModelBuilderTests: XCTestCase {
 
         XCTAssert(responseString == "✅ Hello \(name) ✅")
     }
+
+    func testCreateRequestHandlerWithAction() throws {
+        let transformer = EmojiMediator(emojis: "✅")
+        let component = ActionHandler()
+            .withEnvironment(Connection(state: .open), for: \.connection)
+        let requestHandler = SharedSemanticModelBuilder.createRequestHandler(with: component,
+                                                                             guards: [],
+                                                                             responseModifiers: [ { transformer } ])
+        let name = "Craig"
+        let request = RESTRequest(Vapor.Request(application: app, on: app.eventLoopGroup.next())) { _ in name }
+        let response = try requestHandler(request).wait()
+        let responseString = try XCTUnwrap(response as? String)
+        XCTAssertEqual(responseString, "✅ Hello \(name) ✅")
+    }
```

Comment:
I think you should add a test on how `requestHandler` behaves when `.final` or `.end` is returned.